command/agent/agent.go

```diff
@@ -724,10 +724,16 @@ func (a *Agent) Stats() map[string]map[string]string {
 	return stats
 }
 
+// Reload handles configuration changes for the agent. Provides a method that
+// is easier to unit test, as this action is invoked via SIGHUP.
 func (a *Agent) Reload(newConfig *Config) error {
 	if a.config != nil && newConfig.TLSConfig != nil {
-		if err := a.config.SetTLSConfig(newConfig.TLSConfig); err != nil {
-			return err
+
+		// If the agent is already running with TLS enabled, we need to only reload
+		// its certificates. In a later PR, we will introduce the ability to reload
```

Comment:
Change the in a later PR to:

`// TODO(chelsea)`. so you can find it. Comments should be long lived